### PR TITLE
made docker-push paralelle

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ROOTDIR := $(shell pwd)
 OUTPUT_DIR ?= $(ROOTDIR)/_output
+# Kind of a hack to make sure _output exists
+z := $(shell mkdir -p $(OUTPUT_DIR))
```
Comment (rickducott) [inline Makefile]: Can remove L103 now

File: Makefile [hunk 2]
```diff
@@ -180,10 +182,15 @@ gateway: $(OUTPUT_DIR)/gateway-linux-amd64
 $(OUTPUT_DIR)/Dockerfile.gateway: $(GATEWAY_DIR)/cmd/Dockerfile
 	cp $< $@
 
-gateway-docker: $(OUTPUT_DIR)/gateway-linux-amd64 $(OUTPUT_DIR)/Dockerfile.gateway
+.PHONY: gateway-docker
+gateway-docker: $(OUTPUT_DIR)/.gateway-docker
+
+$(OUTPUT_DIR)/.gateway-docker: $(OUTPUT_DIR)/gateway-linux-amd64 $(OUTPUT_DIR)/Dockerfile.gateway
```
Comment (rickducott) [inline Makefile]: it's not obvious to me that these `.` targets are better... I think I get the intent (reduce redundant work) but this means in development you also have to pass `-B`

Comment (rickducott) [reply Makefile]: I guess we need it so that the push step doesn't re-execute compile 👍

**Comment by yuval-k** [comment]
closing 
cc @EItanya

**Pull request closed without merging**